src/StatusQ/Components/StatusItemSelector.qml

```diff
@@ -0,0 +1,140 @@
+
+    property string icon
+    property int iconSize: 18
+    property string title
+    property string defaultItemText
+    property url defaultItemImageSource: ""
+    property var popupItem
+    property ListModel itemsModel: ListModel { }
+    property string andOperatorText: "and"
+    property string orOperatorText: "or"
+
+    enum Operators {
+        None,
+        And,
+        Or
+    }
+
+    function addItem(text, imageSource, operator) {
+        itemsModel.insert(itemsModel.count, { "text": text, "imageSource": imageSource.toString(), "operator": operator })
+    }
+
+    color: Theme.palette.baseColor4
+    implicitHeight: columnLayout.implicitHeight + columnLayout.anchors.topMargin + columnLayout.anchors.bottomMargin
+    radius: 16
+    clip: true
+    ColumnLayout {
+        id: columnLayout
+        anchors.top: parent.top
+        anchors.topMargin: 12
+        anchors.bottomMargin: anchors.topMargin
+        anchors.left: parent.left
+        anchors.leftMargin: 16
+        anchors.right: parent.right
+        anchors.rightMargin: 16
+        spacing: 12
+        clip: true
+        RowLayout {
+            id: headerRow
+            spacing: 8
+            Image {
+                sourceSize.width: width || undefined
+                sourceSize.height: height || undefined
+                fillMode: Image.PreserveAspectFit
+                mipmap: true
+                antialiasing: true
+                width: root.iconSize
+                height: width
+                source: root.icon
+            }
+            StatusBaseText {
+                Layout.alignment: Qt.AlignVCenter
+                text: root.title
+                color: Theme.palette.directColor1
+                font.pixelSize: 17
+            }
+        }        
+        Flow {
+            id: flow
+            Layout.fillWidth: true
+            spacing: 6
+            StatusListItemTag {
+                visible: itemsModel.count === 0
+                title: root.defaultItemText
+                image.source: root.defaultItemImageSource
+                color: Theme.palette.baseColor2
+                closeButtonVisible: false
+                titleText.color: Theme.palette.baseColor1
+                titleText.font.pixelSize: 15
+            }
+            Repeater {
+                model: itemsModel
+                RowLayout {
+                    spacing: flow.spacing
+                    StatusBaseText {
+                        function operatorTextFormat(operator) {
+                            switch(operator)
+                            {
+                            case StatusItemSelector.Operators.And:
+                                return root.andOperatorText
+                            case StatusItemSelector.Operators.Or:
+                                return root.orOperatorText
+                            case StatusItemSelector.Operators.None:
+                                return ""
+                            }
```

Comment:
I will move the `Operators` enum to a common file like `Utils` and this function to the toplevel but private since it is specific for this component. Thanks for the suggestion!